lib/spack/spack/test/cmd/gpg.py

```diff
@@ -48,134 +60,103 @@ def has_gnupg2():
         return False
 
 
-@pytest.mark.usefixtures('testing_gpg_directory')
```

Comment:
@mathstuf It's done by pytest magic by what they call the *funcargs* dependency injection mechanism. The function that you register as a fixture with the decorator:
```python
@pytest.fixture()
def testing_gpg_directory(...):
    ...
```
acts as a factory for a fixture object. The dependency injection mechanism kicks in when in a test function you declare a function argument with the same name as a registered fixture:
```python
def test_gpg(testing_gpg_directory):
    ...
```

What you have done is fine either, the only difference being that you don't have direct access to the created object.